be/src/olap/rowset/rowset.cpp

```diff
@@ -35,6 +34,12 @@ Rowset::Rowset(const TabletSchema* schema, const FilePathDesc& rowset_path_desc,
         Version version = _rowset_meta->version();
         _is_cumulative = version.first != version.second;
     }
+    // build schema from RowsetMeta.tablet_schema or Tablet.tablet_schema
+    if (_rowset_meta->tablet_schema() != nullptr) {
```

Comment:
I find there are some logic that check whether the rowset has tablet schema. Why not add tablet meta to all rowset during data_dir.load()?
Then we do not need deal with these if else logic.